lib/compiler-cranelift/src/func_environ.rs

```diff
@@ -119,7 +125,9 @@ pub struct FuncEnvironment<'module_environment> {
     /// The memory styles
     memory_styles: &'module_environment PrimaryMap<MemoryIndex, MemoryStyle>,
 
-    /// The table styles
+    /// Cranelift tables we have created to implement Wasm tables.
+    tables: SecondaryMap<TableIndex, Option<TableData>>,
```

Comment:
Why this is an option?

It seems to me it should be:

```suggestion
    tables: SecondaryMap<TableIndex, TableData>,
```

Reply:
The reason, I think, is that there's no sensible default value for `TableData`, and creating a new `SecondaryMap` (both from `new` or `default`) [requires the value type to implement `Default`](https://docs.rs/cranelift-entity/latest/cranelift_entity/struct.SecondaryMap.html#method.new). Perhaps, not using `SecondaryMap`s might be viable as well.